jsoc/gsoc/generativeai.md

```diff
@@ -0,0 +1,107 @@
+- **Expected Outcomes:** 
+  - Expansion of the embedded knowledge base
+  - At least a +10% improvement on the golden Q&A dataset
+
+### Project 6: Native Julia Implementation of ColBERT v2 for Document Retrieval
+
+**Description:** This project involves translating [ColBERT v2](https://github.com/stanford-futuredata/ColBERT), a document retrieval and re-ranking framework, to run natively in Julia. The goal would be to integrate it with AIHelpMe.jl, improving performance and reducing costs.
+
+- **Mentor:** [Jan Siml](https://github.com/svilupp), [Cameron Pfiffer](https://github.com/cpfiffer)  -- looking for co-mentors?
+- **Difficulty**: Hard
+- **Duration**: 350 hours
+- **Suggested Skills and Background**: 
+  - Familiarity with transformer architecture and Flux.jl (/Transformers.jl)
+  - Experience in document retrieval and re-ranking algorithms
+- **Expected Outcomes:** 
+  - Successful translation of ColBERT v2 to native Julia
+  - Integration with AIHelpMe.jl
+  - Registration of the working package
+
+### Project 7: PromptCompiler.jl - Declarative Interface for GenAI Applications
+
+**Description:** The goal of PromptCompiler.jl is to port [DSPy](https://github.com/stanfordnlp/dspy) to Julia, offering a more user-friendly and efficient way to write complex Generative AI pipelines and programs, leveraging Julia's metaprogramming capabilities.
+
+- **Mentor:** [Jan Siml](https://github.com/svilupp), [Cameron Pfiffer](https://github.com/cpfiffer)  -- looking for co-mentors?
+- **Difficulty**: Medium
+- **Duration**: 175 hours
+- **Suggested Skills and Background**: 
+  - Familiarity with prompt engineering
+  - Ability to read Python code (for [DSPy](https://github.com/stanfordnlp/dspy))
+  - Ideally, experience with metaprogramming in Julia
+- **Expected Outcomes:** 
+  - Porting of DSPy to Julia
+  - Development of at least 2 optimizers and 3 modules
+  - Registration of the working
+
+### Project 8: JuliaEval - Building Julia Coding Challenges for Generative AI Fine-Tuning
+
+**Description:** Inspired by [HumanEval](https://github.com/openai/human-eval), JuliaEval aims to create a comprehensive set of Julia coding challenges, along with their solutions, akin to HumanEval. This project is crucial for fine-tuning Generative AI models specifically in the Julia language. It will not only enhance the capabilities of these models in Julia coding tasks but also serve as a valuable resource for the broader community to incorporate into their training pipelines.
+
+- **Mentor:** [Jan Siml](https://github.com/svilupp), [Cameron Pfiffer](https://github.com/cpfiffer) -- looking for co-mentors?
+- **Difficulty**: Medium
+- **Duration**: 175 hours
+- **Suggested Skills and Background**: 
+  - Proficiency in Julia programming
+- **Expected Outcomes:** 
+  - Development of a robust set of Julia coding challenges, covering a wide range of difficulty levels and topics
+  - Comprehensive solutions and explanations for each challenge
```

Comment:
```suggestion
```

This has been done with https://github.com/oolong-dev/OolongEval.jl/tree/add_human_eval 

I may still need a day or two to add more benchmark results and polish the docs.

Overall I think we may replace it with some other proposals.